### PR TITLE
[Feature][Connector-V2] Kylin source connector

### PR DESCRIPTION
**Pull request by 531651225**
## Purpose of this pull request
refer to https://github.com/apache/incubator-seatunnel/issues/3018
add Kylin source Connector
- hive support datatype
https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Types
- kylin jdbc
https://kylin.apache.org/cn/docs31/howto/howto_jdbc.html
### tips
Tip: Since  need to configure the kylin cube model  and build the data in the kylin UI to complete the data preparation, it seems that you cannot add e2e to the program. refer to https://kylin.apache.org/cn/docs/tutorial/create_cube.html
图片

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/kylin/KylinTypeMapper.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    @SuppressWarnings("checkstyle:MagicNumber")
+    @Override
+    public SeaTunnelDataType<?> mapping(ResultSetMetaData metadata, int colIndex) throws SQLException {
+        String kylinType = metadata.getColumnTypeName(colIndex).toUpperCase();
+        int precision = metadata.getPrecision(colIndex);
+        int scale = metadata.getScale(colIndex);
+        switch (kylinType) {
+            case KYLIN_BOOLEAN:
+                return BasicType.BOOLEAN_TYPE;
+            case KYLIN_TINYINT:
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/kylin/KylinTypeMapper.java:69]: We have `short` type maybe you can use it to represent `tinyint`

**Comment by Hisoka-X** [comment]
Please add e2e in https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e

**Comment by 531651225** [comment]
> Please add e2e in https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e

 Since need to configure the kylin cube model and build the data in the kylin UI to complete the data preparation, it seems that you cannot add e2e to the program. so I  added a tips to describe this

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/kylin/KylinTypeMapper.java (hunk 1 above)
Comment (531651225) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/kylin/KylinTypeMapper.java:69]: > We have `short` type maybe you can use it to represent `tinyint`

thinks your good advice，i have fixed it in new commit

**Review by Hisoka-X — DISMISSED** [review]
LGTM @ic4y PTAL.

**Review by TaoZex — DISMISSED** [review]

**Comment by EricJoy2048** [comment]
Please fix the CI problems.

**Comment by 531651225** [comment]
> Please fix the CI problems.
thinks, i have fix it. PTAL.

**Review by TaoZex — APPROVED** [review]

**Comment by ic4y** [comment]
Please add e2e test

**Comment by 531651225** [comment]
> Please add e2e test
Since need to configure the kylin cube model and build the data in the kylin UI to complete the data preparation, it seems that you cannot add e2e to the program. so I added a tips to describe this

**Comment by ic4y** [comment]
> > Please add e2e test
> > Since need to configure the kylin cube model and build the data in the kylin UI to complete the data preparation, it seems that you cannot add e2e to the program. so I added a tips to describe this

Got it, if e2e testing is very difficult. You can comment below your local testing process in the form of pictures, etc. This will make it easier to troubleshoot later if there is a problem.

File: docs/en/connector-v2/source/Jdbc.md [hunk 2]
```diff
@@ -100,6 +100,7 @@ there are some reference value for params above.
 | oracle     | oracle.jdbc.OracleDriver                     | jdbc:oracle:thin:@localhost:1521/xepdb1                      | https://mvnrepository.com/artifact/com.oracle.database.jdbc/ojdbc8 |
 | gbase8a    | com.gbase.jdbc.Driver                        | jdbc:gbase://e2e_gbase8aDb:5258/test                         | https://www.gbase8.cn/wp-content/uploads/2020/10/gbase-connector-java-8.3.81.53-build55.5.7-bin_min_mix.jar |
 | starrocks  | com.mysql.cj.jdbc.Driver                     | jdbc:mysql://localhost:3306/test                             | https://mvnrepository.com/artifact/mysql/mysql-connector-java |
+| kylin      | org.apache.kylin.jdbc.Driver                 | jdbc:kylin://localhost:7070/kylin_project_name               | https://mvnrepository.com/artifact/org.apache.kylin/kylin-jdbc|
```
Comment (EricJoy2048) [inline docs/en/connector-v2/source/Jdbc.md:103]: Please add `changed log` reference https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/connector-v2/source/Redis.md#next-version

**Pull request closed without merging**